pkg/steps/artifacts_test.go

```diff
@@ -712,3 +719,108 @@ func TestAddPodUtils(t *testing.T) {
 	}
 	testhelper.CompareWithFixture(t, base)
 }
+
+func aPod() *coreapi.Pod {
+	return &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      "p",
+			Namespace: "ns",
+		},
+	}
+}
+
+func TestWaitForConditionOnObject(t *testing.T) {
+	containerName := "c"
+	podName := "p"
+	ns := "ns"
+
+	evaluateFunc := func(obj runtime.Object) (bool, error) {
+		switch pod := obj.(type) {
+		case *coreapi.Pod:
+			for _, container := range pod.Status.ContainerStatuses {
+				if container.Name == containerName {
+					if container.State.Running != nil || container.State.Terminated != nil {
+						return true, nil
+					}
+					break
+				}
+			}
+		default:
+			return false, fmt.Errorf("pod/%v ns/%v got an event that did not contain a pod: %v", podName, ns, obj)
+		}
+		return false, nil
+	}
+
+	testCases := []struct {
+		name          string
+		expected      error
+		client        watchingclient.Client
+		identifier    ctrlruntimeclient.ObjectKey
+		list          ctrlruntimeclient.ObjectList
+		into          ctrlruntimeclient.Object
+		evaluate      evaluator
+		timeout       time.Duration
+		containerName string
+		objectFunc    func(client ctrlruntimeclient.Client)
+	}{
+		{
+			name:       "happy path: pod",
+			client:     fakectrlruntimeclient.NewFakeClient(aPod()),
+			identifier: ctrlruntimeclient.ObjectKey{Name: podName, Namespace: ns},
+			list:       &coreapi.PodList{},
+			into:       &coreapi.Pod{},
+			timeout:    time.Second,
+			evaluate:   evaluateFunc,
+			objectFunc: func(client ctrlruntimeclient.Client) {
+				_ = wait.Poll(100*time.Millisecond, time.Second,
```

Comment:
Please somehow handle the error, e.G. by passing in a `*testing.T`

Reply:
I have an impression that we can call `t.errorf` on in the main thread.
I will try to channel up the error.

Reply:
No, you can call `Errorf` anywhere, you just cannot call `Fatalf` from a goroutine

Reply:
Updated.